### PR TITLE
Loading screen when loading, saving, or creating a backup

### PR DESCRIPTION
**Pull request by budak7273**
Thanks @virusek20 for a lot of help on this one. It was a lot more work than I thought it would be.

@Goz3rr please look this one over extra closely; file loading and saving are now async to make this screen possible and I'm worried that I broke something without realizing it by doing this.

### CODE REVIEW

File: SatisfactorySaveEditor/ViewModel/MainViewModel.cs [hunk 1]
```diff
@@ -249,23 +270,28 @@ private void Save(bool saveAs)
                     HasUnsavedChanges = false;
                     RaisePropertyChanged(() => FileName);
                     AddRecentFileEntry(dialog.FileName);
+                    this.IsBusy = false;
```
Comment (Goz3rr) [inline SatisfactorySaveEditor/ViewModel/MainViewModel.cs]: Where is this set to true?

File: SatisfactorySaveEditor/ViewModel/MainViewModel.cs [hunk 2]
```diff
@@ -249,23 +270,28 @@ private void Save(bool saveAs)
...
-                saveGame.Save();
+                this.IsBusy = true;
+                await Task.Run(() => saveGame.Save());
+                this.IsBusy = false;
                 HasUnsavedChanges = false;
             }
         }
 
-        private void AutoBackupIfEnabled()
+        private void AutoBackupIfEnabledAsync()
```
Comment (Goz3rr) [inline SatisfactorySaveEditor/ViewModel/MainViewModel.cs]: You put Async in the name but it's not actually async?

File: SatisfactorySaveEditor/ViewModel/MainViewModel.cs [hunk 3]
```diff
@@ -209,9 +227,10 @@ private void Cheat(ICheat cheat)
         /// </summary>
         /// <param name="saveAs">If the save operation is Save As (unused)</param>
         /// <returns>True if saveGame is NOT null, false otherwise</returns>
-        private bool CanSave(bool saveAs)
+        private bool CanSave(object ignoreThis_NeededForAsyncThing)
```
Comment (Goz3rr) [inline SatisfactorySaveEditor/ViewModel/MainViewModel.cs]: Just use an underscore for unused names

File: SatisfactorySaveEditor/ViewModel/MainViewModel.cs [hunk 4]
```diff
@@ -369,7 +397,7 @@ private void Open(string fileName)
         {
             if (!string.IsNullOrWhiteSpace(fileName))
             {
-                LoadFile(fileName);
+                var _supressAwaitWarning = LoadFile(fileName);
```
Comment (Goz3rr) [inline SatisfactorySaveEditor/ViewModel/MainViewModel.cs]: Use underscore again

File: SatisfactorySaveEditor/ViewModel/MainViewModel.cs (hunk 2 above)
Comment (budak7273) [reply SatisfactorySaveEditor/ViewModel/MainViewModel.cs]: Resolved in next commits

**Pull request merged**